### DISCUSSION TITLE
why is jquery automatically chunked?

### DISCUSSION THREAD

**Discussion by jrochkind** [ElMassimo/vite_ruby#284] [Q&A]
I am loving vite and vite-ruby as I experiment with it and learn more about it, thank you for it!

I have a question that I'm not sure if it's about default vite (or rollup?) or vite-ruby specifically...

Vite seems to be "chunking" jquery separately by default?  Without doing any custom config, the build process is outputting a separate jquery chunk. 

But if I experiment with specifying manualChunks....

```
build: {
   rollupOptions: {
     output: {
       manualChunks: {
         "something": ["something"]
       }
    }
}
```

Now jquery is no longer separately chunked unless I explicitly include it in manualChunks. But without specifying `manualChunks` at all, jQuery was already chunked?

Is there some default manualChunks configuration that vite/vite-ruby/rollup has, where jquery is already in there if unspecified? Googling I can't find any mention of this.  I'm just curious to understand what is going on. 

I import jquery simply like `import $ from jQuery`

**Comment by ElMassimo** [comment]
Vite Ruby does not add any defaults related to chunking, and [Vite doesn't add any defaults since 2.8](https://vitejs.dev/guide/build.html#chunking-strategy).

It's either related to the structure of your application (with Rollup deciding that's the best approach), or with the way jQuery is packaged (esm/cjs heuristics in Vite or Rollup).
